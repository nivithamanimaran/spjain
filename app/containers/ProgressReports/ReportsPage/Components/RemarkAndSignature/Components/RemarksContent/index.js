import React from 'react';
import { makeStyles, Box, Typography, Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';

const useStyles = makeStyles(theme => ({
  subHeadings: {
    color: theme.palette.KenColors.neutral61,
    fontSize: '13px',
  },
  text: {
    color: theme.palette.KenColors.neutral103,
    fontSize: '13px',
    fontWeight: 'bold',
  },
  remarkText: {
    color: theme.palette.KenColors.neutral61,
    fontSize: '13px',
    whiteSpace: 'pre-line',
  },
  box: {
    backgroundColor: theme.palette.KenColors.tertiaryBlue40,
    padding: '16px',
    borderRadius: '6px',
    // minHeight: '100vh',
    height: '100%',
  },
  headings: {
    color: theme.palette.KenColors.tertiaryBlue502,
    fontSize: '12px',
    paddingBottom: '16px',
  },
}));

const RemarksContent = props => {
  const { data, title } = props;
  const classes = useStyles();
  const { t } = useTranslation();

  return (
    <Box className={classes.box} data-testid="remarks-content">
      <Typography
        className={classes.headings}
        data-testid="remarks-content-title"
      >
        {title}
      </Typography>

      <>
        {data &&
          Array.isArray(data) &&
          data.map((item, index) => {
            return (
              <>
                {(item.value || item) && (
                  <Grid item xs={12}>
                    <Typography
                      className={classes.remarkText}
                      data-testid="remarks-content-text"
                    >
                      {item.value || item}
                    </Typography>
                  </Grid>
                )}
              </>
            );
          })}
      </>
    </Box>
  );
};
export default RemarksContent;
