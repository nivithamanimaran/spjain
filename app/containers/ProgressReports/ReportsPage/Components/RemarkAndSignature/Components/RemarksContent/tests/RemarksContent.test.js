import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import RemarksContent from '../index';
import ThemeProvider from '../../../../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

describe('Remarks Content component', () => {
  const div = document.createElement('div');
  const remarks = [
    { label: 'remark', value: 'Priyanka is a very clever student.' },
  ];
  it('Remarks Content component renders correctly without props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <RemarksContent />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <RemarksContent />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('remarks-content')).toMatchSnapshot();
  });
  it('Remarks Content component renders correctly with props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <RemarksContent />
        </Router>
      </ThemeProvider>,
      div
    );

    const { queryByTestId } = render(
      <ThemeProvider>
        <Router>
          <RemarksContent title={'title'} data={remarks} />
        </Router>
      </ThemeProvider>
    );

    const title = queryByTestId('remarks-content-title');
    expect(title).toBeTruthy();
    expect(title.textContent).toBe('title');

    const text = queryByTestId('remarks-content-text');
    expect(text).toBeTruthy();
    expect(text.textContent).toBe('Priyanka is a very clever student.');
  });
});
