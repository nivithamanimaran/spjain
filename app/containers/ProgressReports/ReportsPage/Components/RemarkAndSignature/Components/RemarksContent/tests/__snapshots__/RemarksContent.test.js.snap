// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remarks Content component Remarks Content component renders correctly without props 1`] = `
<div
  class="MuiBox-root MuiBox-root-7 makeStyles-box-4"
  data-testid="remarks-content"
>
  <p
    class="MuiTypography-root makeStyles-headings-5 MuiTypography-body1"
    data-testid="remarks-content-title"
  />
</div>
`;
